packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1137,127 +1137,176 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
+    // The plugin doesn't implement this platform nor reference a default implementation.
+    return (null, null);
+  }
+  String? implementsPackage = plugin.implementsPackage;
+  final bool hasInlineDartImplementation =
+      plugin.pluginDartClassPlatforms[platformKey] != null &&
+          plugin.pluginDartClassPlatforms[platformKey] != 'none';
+  if (implementsPackage == null || implementsPackage.isEmpty) {
+    // Plugin does not serve as implementation for another app-facing package.
+    if (defaultImplPluginName == null && !hasInlineDartImplementation) {
+      // Skip native inline PluginPlatform implementation.
+      return (null, null);
+    }
+    if (defaultImplPluginName != null) {
+      return (null, defaultImplPluginName);
+    } else {
+      // An app-facing package (i.e., one with no 'implements') with an
+      // inline implementation should be its own default implementation.
+      // Desktop platforms originally did not work that way, and enabling
+      // it unconditionally would break existing published plugins, so
+      // only treat it as such if either:
+      // - the platform is not desktop, or
+      // - the plugin requires at least Flutter 2.11 (when this opt-in logic
+      //   was added), so that existing plugins continue to work.
+      // See https://github.com/flutter/flutter/issues/87862 for details.
+      final bool isDesktop = platformKey == 'linux' || platformKey == 'macos' || platformKey == 'windows';
+      final semver.VersionConstraint? flutterConstraint = plugin.flutterConstraint;
+      final semver.Version? minFlutterVersion = flutterConstraint != null &&
+          flutterConstraint is semver.VersionRange ? flutterConstraint.min : null;
+      final bool hasMinVersionForImplementsRequirement = minFlutterVersion != null &&
+          minFlutterVersion.compareTo(semver.Version(2, 11, 0)) >= 0;
+      if (!isDesktop || hasMinVersionForImplementsRequirement) {
+        // Dart plugin serves as its own default implementation.
+        implementsPackage = plugin.name;
+        defaultImplPluginName = plugin.name;
+        return (implementsPackage, defaultImplPluginName);
+      } else {
+        // If the inline dart implementation is desktop and it's flutter min
+        // constraint isn't fulfilled, it isn't eligible for auto-registration.
+        return (null, null);
+      }
+    }
+  }
+
+  if (!hasInlineDartImplementation) {
+    // If there's no Dart implementation, there's nothing to register.
+    return (null, null);
+  }
+
+  // If it hasn't been skipped, it's a candidate for auto-registration,
+  // means it can serve as implementation for [implementsPackage],
+  // so add it as implementation candidate.
+  return (implementsPackage, null);
+}
+
+/// Get the resolved plugin name serving as implementation for a plugin.
+Plugin? _resolveImplementationOfPlugin(
+  String platformKey,
+  MapEntry<String, List<Plugin>> implCandidatesEntry, {
```

Comment:
What does "Entry" indicate in this variable name?

Also, it's not clear from the doc comment what the keys are in the map.

Reply:
Good point, we should split that when handing this over to the method. Resolved in 4e083b834b4f6107f5fdad38fdd0a330e48e0bc4